src/Umbraco.Core/Services/PackagingService.cs

```diff
@@ -435,6 +454,50 @@ public IEnumerable<IContentType> ImportContentTypes(XElement element, bool impor
             return list;
         }
 
+        private Dictionary<string, int> CreateContentTypeFolderStructure(IEnumerable<XElement> unsortedDocumentTypes)
+        {
+            var importedFolders = new Dictionary<string, int>();
+            foreach (var documentType in unsortedDocumentTypes)
+            {
+                var foldersAttribute = documentType.Attribute("Folders");
+                if (foldersAttribute != null)
+                {
+                    var alias = documentType.Element("Info").Element("Alias").Value;
+                    var folders = foldersAttribute.Value.Split('/');
+                    var rootFolder = HttpUtility.UrlDecode(folders[0]);
+                    var current = _contentTypeService.GetContentTypeContainer(rootFolder, 1);
+
+                    if (current == null)
+                    {
+                        var rootFolderId = _contentTypeService.CreateContentTypeContainer(-1, rootFolder).Result;

```

Comment:
This result should probably be checked if it was successful or not before calling `.Result`